internal/pkg/admission/policy-server-ca-secret.go

```diff
@@ -19,30 +20,33 @@ type generateCAFunc = func() (*admissionregistration.CA, error)
 type pemEncodeCertificateFunc = func(certificate []byte) ([]byte, error)
 type generateCertFunc = func(ca []byte, commonName string, extraSANs []string, CAPrivateKey *rsa.PrivateKey) ([]byte, []byte, error)
 
+// reconcileCASecret reconcile the given secret.
+// It updates the secret if it already exists. Otherwise, it creates the secret.
```

Comment:
I'm still going through the PR, why should we call this method? Regenerating the CA is pretty expensive, because lots of things have to be recreated and rolled out. This should be done only when the CA expires